src/gt4py/next/program_processors/runners/gtfn_cpu.py

```diff
@@ -58,6 +73,22 @@ def extract_connectivity_args(
     return args
 
 
+# TODO(tehrengruber): We shouldn't do this. See `GTFNExecutor.use_caching` for a comment.
+def compilation_hash(otf_closure: stages.ProgramCall) -> int:
+    """Given closure compute a hash uniquely determining if we need to recompile."""
+    offset_provider = otf_closure.kwargs["offset_provider"]
+    return hash(
+        (
+            otf_closure.program,
+            # TODO(tehrengruber): as the frontend types contain lists they are
+            #  not hashable. As a workaround we just use content_hash here.
```

Comment:
If this comment only explains why `content_hash` is used here, it does not need to be a TODO. Else it should contain something like "make frontent types hashable" or at least "verify this is the best solution".